mods/default/mapgen.lua

```diff
@@ -1714,6 +1718,8 @@ function default.register_decorations()
 		schematic = minetest.get_modpath("default") .. "/schematics/jungle_log.mts",
 		flags = "place_center_x",
 		rotation = "random",
+		spawn_by = {"default:dirt_with_rainforest_litter", "default:dirt"},
```

Comment:
Hmm, why spawn by 'dirt'? All other logs spawn by the surface node only.

Reply:
I went by the nodes listed in place_on.. dirt is included there. I did wonder about it myself, but if you can place the schematic on a node, then it should surely count when checking surroundings?

Reply:
I see, dirt is in 'on place' for the submerged swamp jungletrees. However if the logs do not force-place then they won't appear in water anyway. Besides, y_min is 1 for logs so they wouldn't appear in water anyway.
Current master code:
```
		name = "default:jungle_log",
		deco_type = "schematic",
		place_on = {"default:dirt_with_rainforest_litter", "default:dirt"},
		sidelen = 80,
		fill_ratio = 0.005,
		biomes = {"rainforest", "rainforest_swamp"},
		y_max = 31000,
		y_min = 1,
```

Reply:
So 'dirt' should not have been in 'place on' in current code either.

Reply:
I have a vague memory of you saying it was due to rainforest_swamp, but don't really remember. I can remove it from both if you want.

Reply:
Thanks please do.